### PR TITLE
Add back resolve x

### PR DESCRIPTION
**Pull request by AndrewNolte**
Adds back COCOTB_RESOLVE_X with the old values, while using low, high, random, and error for the user-passed literals.

### CODE REVIEW

File: src/cocotb/types/logic_array.py
Comment (ktbarrett) [inline src/cocotb/types/logic_array.py:357]: Can't save it. What happens if they get the integer value again and change the resolve method?

**Review by ktbarrett — COMMENTED** [review]
WRT to "random" is the intent that it should be different on every cast or just once?

Also, we should consider since this changes the value if we should save the value and invalidate the other representations.

Also, should the start with integer representation in that case? Otherwise if you do:
```python
>>> a = LogicArray("01XZ")
>>> str(a)
'01XZ'
>>> int(a)
4
>>> str(a)
'0100'
```

IIRC the value was updated in the constructor if there were any Xs in BinaryValue.

File: src/cocotb/types/logic_array.py
Comment (imphil) [inline src/cocotb/types/logic_array.py:355]: Apart from typing, a check for valid values of `resolve` would be nice. Or just use an enum (maybe even preferred).

File: src/cocotb/types/logic_array.py
Comment (imphil) [inline src/cocotb/types/logic_array.py:31]: That's the motivation to change the naming here? On a green field, I'd also prefer lower case and `error` over `VALUE_ERROR`, but given where we are today: is the gain big enough to justify the inconsistency between the environment variable and the function argument? I don't think so.

**Review by imphil — COMMENTED** [review]
Thanks for adding back this feature! Before merging, can we also get a test and documentation added?

File: src/cocotb/types/logic_array.py
Comment (AndrewNolte) [reply src/cocotb/types/logic_array.py:31]: I agree- they should share the same set of values. Since we removed and then added it, and it's used a bit differently, I'm on board with renaming the environment variable (COCOTB_LOGIC_RESOLVE ?) and also using the new values. Another reason for renaming is that it's not just X that it resolves. I made them shorter and lower case since they can now be passed in as an arg, not just an environment variable. But if the maintainers want to keep the old names/values, I can switch these back.

Comment (ktbarrett) [reply src/cocotb/types/logic_array.py:31]: Maybe a StrEnum?

File: src/cocotb/types/logic_array.py
Comment (AndrewNolte) [reply src/cocotb/types/logic_array.py:355]: I prefer to not force people to import something extra just to pass an argument, which is why I'm of a fan of the literal type in place of enums, and not a fan of how Range() was added to LogicArray. I think StrEnum is a good combination, but is only available in 3.11. I can add a backport though.

**Comment by ktbarrett** [comment]
@cmarqu This definitely needs a once over for documentation.

**Comment by ktbarrett** [comment]
Need to add tests.

File: src/cocotb/types/logic_array.py
Comment (cmarqu) [inline src/cocotb/types/logic_array.py:44]:
```suggestion
    "Throws a :exc:`ValueError` if the :class:`LogicArray` contains non-``0``/``1`` values."
```

File: src/cocotb/types/logic_array.py
Comment (cmarqu) [inline src/cocotb/types/logic_array.py:29]:
```suggestion
    """Resolution behaviors supported when converting a :class:`LogicArray` to an integer.
```

File: src/cocotb/types/logic_array.py
Comment (cmarqu) [inline src/cocotb/types/logic_array.py:61]:
```suggestion
    This exists for backwards-compatibility reasons.
```

**Review by cmarqu — APPROVED** [review]
Small doc things only

**Comment by ktbarrett** [comment]
pre-commit.ci autofix

**Review by cmarqu — APPROVED** [review]

File: src/cocotb/_py_compat.py
Comment (ktbarrett) [inline src/cocotb/_py_compat.py:104]:
```suggestion
# inheriting from (str, Enum) was broken in 3.11 and StrEnum must be used
if sys.version_info >= (3, 11):
```

**Pull request merged**
